### ISSUE TITLE
apparent inference hang in makie plot

### ISSUE THREAD

**Issue by JeffBezanson** [JuliaLang/julia#36897]
See https://github.com/JuliaLang/julia/issues/36891#issuecomment-668141193

Test case:
```
using RDatasets: dataset
using AlgebraOfGraphics, AbstractPlotting, GLMakie
using AlgebraOfGraphics: smooth
mpg = dataset("ggplot2", "mpg");
cols = style(:Displ, :Hwy);
grp = style(color = :Cyl => categorical);
scat = spec(Scatter)
pipeline = cols * scat
newstyle = style(markersize = :Cyl) * spec(markersize = (0.1, 5))
data(mpg) * cols * (scat * newstyle + smooth(span = 0.8)) |> draw
```

```
^C^C^C^C^CWARNING: Force throwing a SIGINT
Internal error: encountered unexpected error in runtime:
InterruptException()
jl_unwrap_unionall at /home/jeff/src/julia-1.5/julia/src/jltypes.c:928
obviously_disjoint at /home/jeff/src/julia-1.5/julia/src/subtype.c:305 [inlined]
jl_type_intersection_env_s at /home/jeff/src/julia-1.5/julia/src/subtype.c:3295
jl_typemap_intersection_node_visitor at /home/jeff/src/julia-1.5/julia/src/typemap.c:321
jl_typemap_intersection_visitor at /home/jeff/src/julia-1.5/julia/src/typemap.c:408
ml_matches at /home/jeff/src/julia-1.5/julia/src/gf.c:2810
jl_matching_methods at /home/jeff/src/julia-1.5/julia/src/gf.c:1891
_methods_by_ftype at ./reflection.jl:858 [inlined]
#216 at ./compiler/abstractinterpretation.jl:57 [inlined]
get! at ./iddict.jl:152
jfptr_getNOT._19474 at /home/jeff/src/julia-1.5/julia/usr/lib/julia/sys.so (unknown line)
^Cabstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:55
abstract_call_known at ./compiler/abstractinterpretation.jl:904
abstract_call at ./compiler/abstractinterpretation.jl:926
abstract_call at ./compiler/abstractinterpretation.jl:911
abstract_eval at ./compiler/abstractinterpretation.jl:1005
typeinf_local at ./compiler/abstractinterpretation.jl:1270
typeinf_nocycle at ./compiler/abstractinterpretation.jl:1326
typeinf at ./compiler/typeinfer.jl:12
typeinf_edge at ./compiler/typeinfer.jl:484
```

Happens in 1.5. Works in 1.4 and master.

**Comment by JeffBezanson** [comment]
Trying this again, I get the BoundsError from #36891 on master but otherwise it works. So unfortunately seems to be specific to 1.5.

**Comment by JeffBezanson** [comment]
Seems to have been caused by 301db971daaeeb627ba768375538e6e7ff36d215. Somehow those definitions trigger some bad case in inference.

**Comment by JeffBezanson** [comment]
Inference issues with that commit have been noticed before: https://github.com/JuliaLang/julia/pull/34048#issuecomment-624594668, #35800.

**Comment by SimonDanisch** [comment]
Seems fixed on master & 1.5.2
